source/cpu/cpu.go

```diff
@@ -274,33 +274,65 @@ func getCPUModel() map[string]string {
 	return cpuModelInfo
 }
 
-// getHypervisor detects the hypervisor on s390x by reading /proc/sysinfo.
-// If the file does not exist, it returns an empty string with no error.
+// getHypervisor detects the hypervisor on s390x by reading /proc/sysinfo
+// and on x86_64/arm64 using the cpuid library.
+// Returns normalized hypervisor string, "unknown" for unknown vendor, or "none" for bare metal.
 func getHypervisor() (string, error) {
-	if _, err := os.Stat("/proc/sysinfo"); os.IsNotExist(err) {
-		return "", nil
+	// try using /proc/sysinfo for s390x
+	if _, err := os.Stat("/proc/sysinfo"); err == nil {
+		hv, discovered := getHypervisorFromProcSysinfo()
+		if discovered {
+			return hv, nil
+		}
```

Comment:
If this does not find anything (`discovered == false`), do we want to fall back to `getHypervisorFromCPUID()`? Probably not, or WDYT?